lib/controller.js

```diff
@@ -641,6 +663,213 @@ controller.tesselFirmwareVerion = function(opts) {
         });
     });
 };
+controller.menu = {
+
+  prepareMenu: function(opts, tessels, resolve, reject, title, callback) {
+    //return new Promise(function(resolve, reject) {
+
+    var rtm = new Menu({
+      //  width: process.stdout.columns - 4,
+      width: 55,
+      x: 1,
+      y: 2,
+      bg: 'red'
+    });
+    rtm.reset();
+    rtm.write(title + '\n');
+    rtm.write('                                                \n');
+
+    // create menu entries
+    for (var i in tessels) {
+      controller.menu.makeMenu(tessels[i], i, rtm);
+    }
+
+    rtm.add('EXIT\n');
+    rtm._draw();
+    if (!opts.menu) {
+      // if this is no test, starting the menu as child process
+      controller.menu.showMenu(rtm, resolve, reject);
+    } else {
+      // because the test needs to resolve the promise ...
+      resolve({
+        opts: opts,
+        tessels: tessels
+      });
+    }
+
+    rtm.once('select', function(label, index) {
+      rtm.close();
+      controller.menu.clear();
+
+      // Identify the Exit command by first letter (all other entries start with numbers/index)
+      if (label[0] !== 'E') {
+        callback(tessels[index].name, index, resolve, reject);
+        // controller.ssh.runSSH(index, opts, tessels, resolve, reject);
+      } else {
+        // going to clear screen and calling exit to resolve promise
+        controller.menu.clear();
+        controller.menu.exit(resolve);
+      }
+    });
+    //  });
+  },
+  makeMenu: function(tessel, index, rtm) {
+    if (tessel.lanConnection) {
+      if (!tessel.lanConnection.authorized) {
+        rtm.add(index + ') ' + tessel.name + ': ' + tessel.lanConnection.ip + ' (not authorized) \n');
+      } else {
+        rtm.add(index + ') ' + tessel.name + ': ' + tessel.lanConnection.ip + ' (authorized) \n');
+      }
+
+    } else if (tessel.usbConnection) {
+      if (!tessel.usbConnection.authorized) {
+        rtm.add(index + ') ' + tessel.name + ': [USB] (not authorized) \n');
```

Comment:
Yes, it's possible some folks will be confused about that. We were initially planning on making `t2 provision` run automatically but ultimately decided against it so that users don't get confused about trying to deploy to LAN devices that they've never connected over USB before.